mixins/grid_layout.go

```diff
@@ -0,0 +1,80 @@
+package mixins
+
+import (
+  "github.com/google/gxui"
+  "github.com/google/gxui/math"
+  "github.com/google/gxui/mixins/base"
+)
+
+type Cell struct {
+  x, y, w, h int
+}
+
+type GridLayoutOuter interface {
+  base.ContainerOuter
+}
+
+type GridLayout struct {
+  base.Container
+
+  outer GridLayoutOuter
+
+  grid []Cell
+  rows int
+  columns int
+}
+
+func (l *GridLayout) Init(outer GridLayoutOuter, theme gxui.Theme) {
+  l.Container.Init(outer, theme)
+  l.outer = outer
+
+  _ = gxui.GridLayout(l)
+}
+
+func (l *GridLayout) LayoutChildren() {
+  s := l.outer.Size().Contract(l.outer.Padding())
+	o := l.outer.Padding().LT()
+
+  children := l.outer.Children()
+
+  cw, ch := s.W / l.columns, s.H / l.rows
+
+  var cr math.Rect
+
+  for i, cell := range l.grid {
+    c := children[i]
+    cm := c.Control.Margin()
+
+    x, y := cell.x * cw, cell.y * ch
+    w, h := x + cell.w * cw, y + cell.h * ch
```

Comment:
Nit: There's no protection or warning here about overlapping cells - not sure what you could do to help that, or if that's worth the effort anyway.